OpenRA.Mods.Common/UpdateRules/Rules/20201213/ReplaceResourceValueModifiers.cs

```diff
@@ -0,0 +1,41 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class ReplaceResourceValueModifiers : UpdateRule
+	{
+		public override string Name { get { return "HarvesterResourceMultiplier and RefineryResourceMultiplier replaced with ResourceValueMultiplier."; } }
+
+		public override string Description
+		{
+			get
+			{
+				return "The HarvesterResourceMultiplier trait has been removed, and the RefineryResourceMultiplier trait renamed to ResourceValueMultiplier.";
```

Comment:
Graion notified me of this change on Discord, while i don't care much about GenAlpha anymore. The HarvesterResourceMultiplier was used for the implementation of Worker Shoes update, which increased the resource dropped by Workers to any Supply Center. Removing the trait would remove this ability, and while i haven't looked in detail, doesn't seem like new trait is able to do that. On the opposite end for RefineryResourceMultiplier was used by Supply Lines upgrade, which increased resource dropped to American Supply Center by any supply unit.

I think rather than keeping 2 traits for this. Adding an actor type check to ResourceValueMultiplier could let us keep this ability. So the trait would sit on the Supply Center on both cases, but you can filter the Worker Shoes case so only modifier only apply to resources dropped by Workers.

Reply:
`INotifyResourceAccepted` has access to everything you need to implement that.

Reply:
Updated to make `Harvester.TickUnload` virtual, so Generals Alpha can do whatever it needs.